app/tests/spec/views/force_auth.js

```diff
@@ -86,44 +87,112 @@ function (chai, $, sinon, View, FxaClient, p, AuthErrors, Relier, Broker,
-          sinon.stub(user, 'signInAccount', function () {
-            done();
-          });
-          view.$('#submit-btn').click();
+      describe('rendering', function () {
+        it('does not print an error message', function () {
+          assert.equal(view.$('.error').text(), '');
+        });
+
+        it('email input is hidden for the Firefox Password manager', function () {
+          assert.equal(view.$('input[type=email]').hasClass('hidden'), 1);
+        });
+
+        it('prefills password', function () {
+          assert.equal(view.$('input[type=password]').val(), 'password');
         });
 
-        it('submits the sign in', function () {
-          var password = 'password';
-          sinon.stub(user, 'signInAccount', function (account) {
-            account.set('verified', true);
-            return p(account);
+        it('user cannot create an account', function () {
+          assert.equal(view.$('a[href="/signup"]').length, 0);
+        });
+      });
+
+      describe('submit', function () {
+        var password;
+
+        beforeEach(function () {
+          password = 'password';
+
+          sinon.stub(view, '_signIn', function (account) {
+            return p();
           });
+
           view.$('input[type=password]').val(password);
 
-          return view.submit()
-            .then(function () {
-              assert.equal(user.signInAccount.args[0][0].get('email'), email);
-              assert.equal(user.signInAccount.args[0][0].get('password'), password);
-            });
+          return view.submit();
         });
-      });
 
-      it('does not print an error message', function () {
-        assert.equal(view.$('.error').text(), '');
+        it('calls view._signIn with the expected data', function () {
+          var account = view._signIn.args[0][0];
+          assert.equal(account.get('email'), email);
+          assert.equal(account.get('password'), password);
+        });
       });
 
-      it('email input is hidden for the Firefox Password manager', function () {
-        assert.equal(view.$('input[type=email]').hasClass('hidden'), 1);
-      });
+      describe('onSignInSuccess', function () {
+        var account;
+
+        beforeEach(function () {
+          account = user.initAccount({
+            email: 'testuser@testuser.com',
+            password: 'password'
+          });
+
+          sinon.spy(broker, 'afterForceAuth');
+          sinon.spy(view, 'navigate');
 
-      it('prefills password', function () {
-        assert.equal(view.$('input[type=password]').val(), 'password');
+          return view.onSignInSuccess(account);
+        });
+
+        it('invokes `afterForceAuth` on the broker', function () {
+          assert.isTrue(broker.afterForceAuth.calledWith(account));
+        });
+
+        it('navigates to the `settings` page and clears the query parameters', function () {
+          assert.isTrue(view.navigate.calledWith('settings', { clearQueryParams: true }));
```

Comment:
Unless I'm missing it, I don't see an equivalent test for `/confirm_reset_password`. Can we add one or is that tricky for some reason?